sharding/tests/test_validator_manager_utils.py

```diff
@@ -104,20 +87,46 @@ def get_colhdr(shard_id, parent_collation_hash, collation_coinbase=t.a0):
             parent_collation_hash, tx_list_root, collation_coinbase,
             post_state_root, receipt_root, sig
         ])
+
+    # register t.k0 as the validators
+    tx = create_contract_tx(chain.head_state, t.k0, mk_validation_code(t.a0))
+    k0_valcode_addr = chain.direct_tx(tx)
+    chain.mine(1)
+
+    tx = call_deposit(chain.head_state, t.k0, DEPOSIT_SIZE, k0_valcode_addr, t.a0)
+    chain.direct_tx(tx)
+    chain.mine(1)
+
+    # create collation header
     shard0_genesis_colhdr_hash = utils.encode_int32(0)
     colhdr = get_colhdr(0, shard0_genesis_colhdr_hash)
     colhdr_hash = utils.sha3(colhdr)
-    assert call_get_shard_head(state, 0) == shard0_genesis_colhdr_hash
-    # register t.k0 as the validators
-    k0_valcode_addr = deploy_contract(state, t.k0, mk_validation_code(t.a0))
-    tx = call_deposit(state, t.k0, deposit_size, k0_valcode_addr, t.a2)
-    deploy_tx(state, tx)
+    assert call_get_shard_head(chain.head_state, 0) == shard0_genesis_colhdr_hash
+
+    # message call test
+    assert call_msg_add_header(chain.head_state, 0, colhdr, t.a0)
+
+    # transaction call test
     # `add_header` verifies whether the colhdr is signed by the current
     # selected validator, using `sample`
-    tx = call_add_header(state, t.k0, 0, colhdr)
-    deploy_tx(state, tx)
-    assert colhdr_hash == call_get_shard_head(state, 0)
+    tx = call_tx_add_header(chain.head_state, t.k0, 0, colhdr)
+    chain.direct_tx(tx)
+    chain.mine(1)
+
+    assert colhdr_hash == call_get_shard_head(chain.head_state, 0)
+
 
 def test_valmgr_addr_in_sharding_config():
     assert sharding_config['VALIDATOR_MANAGER_ADDRESS'] == \
-           utils.checksum_encode(get_valmgr_addr())
+        utils.checksum_encode(get_valmgr_addr())
+
+
+def test_sign():
+    """Test collator.sign(msg_hash, privkey)
+    """
+    msg_hash = utils.sha3('hello')
+    privkey = t.k0
+    assert sign(msg_hash, privkey) == b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x1bz\x05\x13\xf6\xa4\xbb\x0c\xaf<\x87\x95\xa7\xf5\x139\x84\x89\\#\x91\x15\x9dPX\x9e\xc9\x01\x8fp\x14\xd2,\x0c\x97\xd6\xbf\xc9\x11\x9d\xf7Z\x99-\xd3\x05\xc6\xf3\xfc\xfbe\x99c1\xcb\x93K\xf0I,\xd7\xebUB%'
+
+    msg_hash2 = utils.sha3('world')
+    assert sign(msg_hash2, privkey) != b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x1bz\x05\x13\xf6\xa4\xbb\x0c\xaf<\x87\x95\xa7\xf5\x139\x84\x89\\#\x91\x15\x9dPX\x9e\xc9\x01\x8fp\x14\xd2,\x0c\x97\xd6\xbf\xc9\x11\x9d\xf7Z\x99-\xd3\x05\xc6\xf3\xfc\xfbe\x99c1\xcb\x93K\xf0I,\xd7\xebUB%'
```

Comment:
Isn't this `assert` a little bit strange? Except for "world", the `sha3` of any other string would not equal to the `sha3` of "hello".